### PR TITLE
feat(pkger): add ability to export resources by name from cli

### LINKED ISSUE #18733: Provide ability to export by name for all resources
as a user,
I want to export resources by name to a template,
so I don't have to juggle UUIDs myself

**Acceptance Criteria**

- [ ] ability to export via CLI by name and or id

```sh
$ influx export --label=UUID --label="Name of label" ...
```

- [ ] support through the API by extending the `POST /api/v2/packages` with the following:

```json
{
   ...existing,
   "resources": [
       {
            "kind": KIND,
            "resourceID": UUID,
            "resourceName": RES_NAME,
            "rename": NEW_NAME_IN_TEMPLATE
       }
    ]
}
```

resourceName will be the identifer to match by. If an ID and or Name is provided it will match on that. At least 1 of resourceID or resourceName will be required

### PR DESCRIPTION
**Pull request by glinton**
Closes #18733

Export resources by name via the cli.

- [x] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [x] Tests pass
- [x] Documentation updated or issue created (https://github.com/influxdata/docs-v2/issues/1526)

### CODE REVIEW

File: cmd/influxd/launcher/pkger_test.go [hunk 1]
```diff
@@ -1970,6 +1970,7 @@ func TestLauncher_Pkger(t *testing.T) {
 			testStackApplyFn := func(t *testing.T) (pkger.Summary, pkger.Stack, func()) {
 				t.Helper()
 
+				// asdfasfd
```
Comment (hoorayimhelping) [inline cmd/influxd/launcher/pkger_test.go:1973]: it's not that i don't love gibberish...

✂️

File: pkger/clone_resource.go [hunk 2]
```diff
@@ -27,7 +27,7 @@ type NameGenerator func() string
 // ResourceToClone is a resource that will be cloned.
 type ResourceToClone struct {
 	Kind Kind        `json:"kind"`
-	ID   influxdb.ID `json:"id"`
+	ID   influxdb.ID `json:"id,omitempty"`
```
Comment (hoorayimhelping) [inline pkger/clone_resource.go:30]: do you know if any frontend code depends on this? changing how it handles empty _might_ require changes on the javascript side.

File: pkger/clone_resource.go [hunk 3]
```diff
@@ -220,77 +217,255 @@ func (ex *resourceExporter) resourceCloneToKind(ctx context.Context, r ResourceT
...
+		var (
+			hasID  bool
+			filter = influxdb.DashboardFilter{}
+		)
+		if r.ID != influxdb.ID(0) {
+			hasID = true
+			filter.IDs = []*influxdb.ID{&r.ID}
+		}
+
+		dashs, _, err := ex.dashSVC.FindDashboards(ctx, filter, influxdb.DefaultDashboardFindOptions)
```
Comment (hoorayimhelping) [inline pkger/clone_resource.go:270]: pedantic: but should this be `dashes`?

File: pkger/clone_resource.go [hunk 4]
```diff
@@ -220,77 +217,255 @@ func (ex *resourceExporter) resourceCloneToKind(ctx context.Context, r ResourceT
...
 		r.Kind.is(KindNotificationEndpointPagerDuty),
 		r.Kind.is(KindNotificationEndpointSlack):
-		e, err := ex.endpointSVC.FindNotificationEndpointByID(ctx, r.ID)
-		if err != nil {
-			return err
+		var endpoints []influxdb.NotificationEndpoint
+
+		switch {
+		case r.ID != influxdb.ID(0):
+			ndpoint, err := ex.endpointSVC.FindNotificationEndpointByID(ctx, r.ID)
```
Comment (hoorayimhelping) [inline pkger/clone_resource.go:323]: should this be `endpoint`?

can it be called `notifEndpoint` if `endpoint` is taken?

File: pkger/service.go [hunk 5]
```diff
@@ -568,6 +569,7 @@ func ExportWithExistingResources(resources ...ResourceToClone) ExportOptFn {
 	return func(opt *ExportOpt) error {
 		for _, r := range resources {
 			if err := r.OK(); err != nil {
+				// todo: log and continue else append r to opt.Resources??
```
Comment (hoorayimhelping) [inline pkger/service.go:572]: does this need to be addressed in this PR?

File: pkger/service.go [hunk 6]
```diff
@@ -736,9 +741,12 @@ func (s *Service) cloneOrgDashboards(ctx context.Context, orgID influxdb.ID) ([]
 }
 
 func (s *Service) cloneOrgLabels(ctx context.Context, orgID influxdb.ID) ([]ResourceToClone, error) {
-	labels, err := s.labelSVC.FindLabels(ctx, influxdb.LabelFilter{
+	filter := influxdb.LabelFilter{
 		OrgID: &orgID,
-	}, influxdb.FindOptions{Limit: 10000})
+	}
+
+	// todo: if this was ever called, it would error, 100 is the limit.
```
Comment (hoorayimhelping) [inline pkger/service.go:748]: does this todo need to be addressed?

File: pkger/service_logging.go [hunk 7]
```diff
@@ -148,7 +148,8 @@ func (s *loggingMW) Export(ctx context.Context, opts ...ExportOptFn) (template *
 			s.logger.Error("failed to export template", zap.Error(err), dur)
 			return
 		}
-		s.logger.Info("failed to export template", append(s.summaryLogFields(template.Summary()), dur)...)
+		// todo: should these be Debug logs?
```
Comment (hoorayimhelping) [inline pkger/service_logging.go:151]: @russorat are you the man to answer this?

**Review by hoorayimhelping — CHANGES_REQUESTED** [review]
No hard blockers, just a couple of questions

File: pkger/clone_resource.go (hunk 2 above)
Comment (glinton) [reply pkger/clone_resource.go:30]: i'm not certain the relationship, but it is required to be omitted when exporting by name (specifying only name in the payload). i suppose the swagger should be updated to note that it is only required when no name is specified and optional if it is specified. ...i don't love how if both are specified, the resource is found by the id and the exported result is renamed to the name provided.

File: pkger/service.go (hunk 5 above)
Comment (glinton) [reply pkger/service.go:572]: probably not, it was a note to investigate if that was necessary.

File: pkger/service_logging.go (hunk 7 above)
Comment (russorat) [reply pkger/service_logging.go:151]: Based on what i see in other parts of the app, i would think these "failed to" logs should be debug.

Comment (glinton) [reply pkger/service_logging.go:151]: the `failed to` seemed to be copypasta. the logs in question are the success logs

https://github.com/influxdata/influxdb/pull/19457/files/53b325c7d0d16ce11723083cd3a0243e481881ce#diff-11efed3cf4ce3c7284b325143fb8b70dR180-R181
```go
s.logger.Info("template dry run successful", fields...)
// ...
s.logger.Info("template apply successful", fields...)
```

Comment (russorat) [reply pkger/service_logging.go:151]: Ah, if they are success logs, i would think they should be Info

File: pkger/service_logging.go [hunk 8]
```diff
@@ -176,6 +177,7 @@ func (s *loggingMW) DryRun(ctx context.Context, orgID, userID influxdb.ID, opts
 			fields = append(fields, zap.Stringer("stackID", opt.StackID))
 		}
 		fields = append(fields, dur)
+		// todo: should these be Debug logs?
```
Comment (hoorayimhelping) [inline pkger/service_logging.go:180]: ✂️ cut these comments

**Review by hoorayimhelping — APPROVED** [review]
Works for me, but I haven't tested the swagger definition with `yarn generate`. is the swagger def good to go?

File: kv/dashboard.go [hunk 9]
```diff
@@ -135,7 +135,9 @@ func filterDashboardsFn(filter influxdb.DashboardFilter) func(d *influxdb.Dashbo
 		}
 	}
 
-	return func(d *influxdb.Dashboard) bool { return true }
+	return func(d *influxdb.Dashboard) bool {
+		return ((filter.OrganizationID == nil) || (*filter.OrganizationID == d.OrganizationID))
```
Comment (glinton) [inline kv/dashboard.go:139]: Hoping this is kosher, it follows the pattern of [labels](https://github.com/influxdata/influxdb/pull/19457/files#diff-b00232c908213b364c39e0c732ed8f1dR139)
```go
func filterLabelsFn(filter influxdb.LabelFilter) func(l *influxdb.Label) bool {
	return func(label *influxdb.Label) bool {
		return (filter.Name == "" || (strings.EqualFold(filter.Name, label.Name))) &&
			((filter.OrgID == nil) || (filter.OrgID != nil && *filter.OrgID == label.OrgID))
	}
}
```

**Pull request merged**